### PR TITLE
Add PayPal redirect URL test 

### PR DESCRIPTION
**Pull request by desbo**
This change adds two tests for the `executePayment` method in `PayPalController`. One mocks a successful payment response and asserts that the redirect is to the post-payment URL (precluding any error parameters), the other tests that payment errors redirect to the appropriate error URL.

@guardian/contributions

### CODE REVIEW

**Review by dominickendrick — APPROVED** [review]
Looks good! Merge once build is fixed

**Comment by desbo** [comment]
It required a couple of extra mocks, and unfortunately making some `val`s `lazy`, but the build is finally green now

**Review by joelochlann — APPROVED** [review]
Nice work

File: test/controllers/PaypalControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import controllers.PaypalController
+import org.scalatest.mockito.MockitoSugar
+import org.scalatestplus.play._
+import play.api.mvc._
+import play.api.test.Helpers.{redirectLocation, _}
+import play.api.test._
+
+class PaypalControllerSpec extends PlaySpec
+  with Results
```
Comment (joelochlann) [inline test/controllers/PaypalControllerSpec.scala]: You can get rid of `Results` and `MockitoSugar` and associated imports

**Pull request merged**